### PR TITLE
Fix rest-server tests

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: Shlomit Koyfman <shlomitk@il.ibm.com>
Fixes https://github.com/fybrik/fybrik/issues/1374

### CODE REVIEW

File: samples/rest-server/datauser/credentialapis.go [hunk 1]
```diff
@@ -66,24 +69,23 @@ func GetCredentials(w http.ResponseWriter, r *http.Request) {
...
+	secretName := chi.URLParam(r, paramSecret)
 
 	// Call kubernetes to get the FybrikApplication CRD
 	err := k8sClient.DeleteSecret(secretName, nil)
 	if err != nil {
 		suberr := render.Render(w, r, ErrRender(err))
 		if suberr != nil {
-			log.Printf(suberr.Error() + " upon " + err.Error())
+			log.Printf(suberr.Error() + " : " + err.Error())
 		}
```
Comment (roytman) [inline samples/rest-server/datauser/credentialapis.go:89]: When a REST call returns without setting an HTTP error, what should be returned to the client?
should we set something like `render.Status(r, http.StatusInternalServerError)`

File: samples/rest-server/datauser/credentialapis.go [hunk 2]
```diff
@@ -99,9 +101,9 @@ func StoreCredentials(w http.ResponseWriter, r *http.Request) {
 
 	log.Println("In StoreCredentials")
 	if k8sClient == nil {
-		suberr := render.Render(w, r, ErrConfigProblem(errors.New("no k8sClient set")))
+		suberr := render.Render(w, r, ErrConfigProblem(clientNotSetError()))
 		if suberr != nil {
-			log.Printf(suberr.Error() + " upon No k8sClient set")
+			log.Print(suberr.Error())
```
Comment (roytman) [inline samples/rest-server/datauser/credentialapis.go:106]: like above

File: samples/rest-server/datauser/credentialapis.go (hunk 1 above)
Comment (shlomitk1) [reply samples/rest-server/datauser/credentialapis.go]: Fixed

File: samples/rest-server/datauser/credentialapis.go (hunk 2 above)
Comment (shlomitk1) [reply samples/rest-server/datauser/credentialapis.go]: Fixed

**Review by roytman — APPROVED** [review]

**Comment by roytman** [comment]
do we need this comment https://github.com/fybrik/fybrik/blob/54d7f0f5f5a32750e1a50a065fe8a62b9048024e/samples/rest-server/datauser/utils.go#L76 ?

**Comment by shlomitk1** [comment]
> do we need this comment
> 
> https://github.com/fybrik/fybrik/blob/54d7f0f5f5a32750e1a50a065fe8a62b9048024e/samples/rest-server/datauser/utils.go#L76
> 
> ?

I guess not...

**Pull request merged**